tutorials/shaders/custom_postprocessing.rst

```diff
@@ -6,167 +6,114 @@ Custom post-processing
 Introduction
 ------------
 
-Godot provides many post-processing effects out of the box, including Bloom, DOF, and SSAO. Sometimes you
-want to write your own custom effect. Here's how you can do so.
-
-Post-processing effects are shaders applied to a frame after Godot rendered it. You first want to render
-your scene into a :ref:`Viewport <class_Viewport>`, then render the ``Viewport``
-inside a :ref:`ViewportTexture <class_ViewportTexture>` and show it on the screen.
+Godot provides many post-processing effects out of the box, including Bloom, DOF, and SSAO. Sometimes you want to write your own custom effect. Here's how you can do so.
 
 The easiest way to implement a custom post-processing shader is to use Godot's built-in ability to read from
 the screen texture. If you're not familiar with this, you should read the :ref:`Screen Reading Shaders
 Tutorial <doc_screen-reading_shaders>` first.
 
-.. note::
-
-    As of the time of writing, Godot does not support rendering to multiple buffers at the same time. Your
-    post-processing shader will not have access to normals or other render passes. You only have
-    access to the rendered frame.
-
 Single pass post-processing
 ---------------------------
 
-You will need a ``Viewport`` to render your scene to, and a scene to render your
-``Viewport`` on the screen. You can use a :ref:`ViewportContainer
-<class_ViewportContainer>` to display your ``Viewport`` on the entire screen or inside
-another :ref:`Control <class_Control>` node.
-
-.. note::
-
-    Rendering using a ``Viewport`` gives you control over
-    how the scene render, including the framerate, and you can use the
-    ``ViewportContainer`` to render 3D objects in a 2D scene.
-
-For this demo, we will use a :ref:`Node2D <class_Node2D>` with a ``ViewportContainer`` and finally a
-``Viewport``. Your **Scene** tab should look like this:
-
-.. image:: img/post_hierarchy1.png
-
-Inside the ``Viewport``, you can have whatever you want. This will contain
-your main scene. For this tutorial, we will use a field of random boxes:
+Post-processing effects are shaders applied to a frame after Godot rendered it. To apply a shader to a frame, create a :ref:`CanvasLayer <class_CanvasLayer>`, and give it a :ref:`ColorRect <class_ColorRect>`. Give it a new :ref:`ShaderMaterial <class_ShaderMaterial>`, and set the ``ColorRect``'s layout to "Full Rect".
```

Comment:
In the RST docs, single newline characters are ignored, so you can split up paragraphs into multiple lines. Note that this is different from the class reference which does care about single newline characters.